### PR TITLE
feat: create admin user on first login

### LINKED ISSUE #1518: Initial signup should create an admin user
Initial login process through CLI should have the user create a new admin account. This account will be automatically grant admin Infra privileges

- change `/v1/setup` to `/v1/signup` to signal it creates an admin account

The existing restrictions for when signup is possible remain unchanged

### PR DESCRIPTION
**Pull request by mxyng**
## Summary

Rework the `setup` workflow to instead run a `sign-up` workflow. Sign-up will allow the first user to immediate create an admin user without running subsequent infra commands.

### Minor refactoring

* Consolidate prompt validation with `survey.Validator`s
* Revert prompt formatting since it does not look very good

## Checklist

- [x] Wrote appropriate unit tests
- [ ] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [ ] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #1518

### CODE REVIEW

File: internal/cmd/login.go [hunk 1]
```diff
@@ -315,19 +319,28 @@ func loginToProvider(provider *api.Provider) (*api.LoginRequestOIDC, error) {
 	}, nil
 }
 
-func runSetupForLogin(client *api.Client) (string, error) {
-	setupRes, err := client.Setup()
+func runSignupForLogin(client *api.Client) (*api.LoginRequestPasswordCredentials, error) {
```
Comment (mxyng) [inline internal/cmd/login.go:322]: User facing messages probably needs some refining. Right now it displays
```
$ infra login localhost
  Welcome to Infra. Set up your admin user.
? Email: test@example.com
? Password: ********
? Confirm Password: ********
  Logged in as test@example.com
```

File: api/signup.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package api
+
+type SignupRequiredResponse struct {
+	Required bool `json:"required"`
+}
+
+type SignupRequest struct {
+	Name     string `json:"email"`
```
Comment (jmorganca) [inline api/signup.go:8]: These two should be the same

I'd recommend `Name` and `name`

File: docs/getting-started/quickstart.md [hunk 3]
```diff
@@ -85,44 +85,10 @@ Use the following command to find the Infra login URL. If you are not using a `L
...
-
-```
-infra login
-```
-
-Select the Infra instance, and login with username `name@example.com`, and the password
-from the previous step.
-
-### 5. Connect your first Kubernetes cluster
+### 3. Connect your first Kubernetes cluster
```
Comment (jmorganca) [inline docs/getting-started/quickstart.md:91]: Awesome to see the number of steps  🪓

File: internal/cmd/login.go (hunk 1 above)
Comment (jmorganca) [reply internal/cmd/login.go:322]: This is spot on, would just replace the `.` with `:`

```
$ infra login localhost Welcome to Infra. Set up your admin user:
```

Comment (mxyng) [reply internal/cmd/login.go:322]: Updated
```
$ infra login localhost:
  Welcome to Infra. Set up your admin user:
? Email: test@example.com
? Password: ********
? Confirm Password: ********
  Logged in as test@example.com
```

**Review by jmorganca — APPROVED** [review]

File: internal/access/signup.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	identity := &models.Identity{
+		Name: name,
+		Kind: models.UserKind,
+	}
+
+	if err := data.CreateIdentity(db, identity); err != nil {
+		return nil, err
+	}
+
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
```
Comment (ssoroka) [inline internal/access/signup.go:49]: Not your bug, but there's another spot we're using `bcrypt.MinCost`, can you change that to `bcrypt.DefaultCost` as well?

**Review by ssoroka — APPROVED** [review]
obviously will be a breaking change for the API.

**Pull request merged**